src/if_statements.py

```diff
@@ -491,7 +476,7 @@ def get_andor_if_statement(
             # An edge-case of our pattern-matching technology: without
             # this, self-loops match the pattern indefinitely, since a
             # self-loop node's conditional edge points to itself.
-            or next_node.is_loop()
+            or next_node.loop
```

Comment:
Is this equivalent? I thought `.loop` meant loop head...

Reply:
It's not equivalent - but I think it's the intended logic? 
Before, `is_loop()` referred to the tail of the loop: the BB with the backedge to the head node. But here, we don't actually mind incorporating the tail of the loop into the compound if -- we want to avoid bringing the *head* into the compound if. 
(I think an alternative would be just using the parents constraint: `len(next_node.parents) > 1 or next_node in next_node.parents`)